src/pages/home/sidebar/SidebarLinks.js

```diff
@@ -156,15 +169,30 @@ class SidebarLinks extends React.Component {
     constructor(props) {
         super(props);
         this.state = {
-            currentlyViewedReportID: props.currentlyViewedReportID,
+            activeReport: {
+                reportID: props.currentlyViewedReportID,
+                hasDraftHistory: lodashGet(props.reportsWithDraft, `${ONYXKEYS.COLLECTION.REPORTS_WITH_DRAFT}${props.currentlyViewedReportID}`, false),
+                lastMessageTimestamp: lodashGet(props.reports, `${ONYXKEYS.COLLECTION.REPORT}${props.currentlyViewedReportID}.lastMessageTimestamp`, 0),
+            },
             orderedReports: [],
             priorityMode: props.priorityMode,
             unreadReports: SidebarLinks.getUnreadReports(props.reports || {}),
         };
     }
 
     static getDerivedStateFromProps(nextProps, prevState) {
-        const shouldReorder = SidebarLinks.shouldReorder(nextProps, prevState.orderedReports, prevState.currentlyViewedReportID, prevState.unreadReports);
+        const isActiveReportSame = prevState.activeReport.reportID === nextProps.currentlyViewedReportID;
+        const lastMessageTimestamp = lodashGet(nextProps.reports, `${ONYXKEYS.COLLECTION.REPORT}${nextProps.currentlyViewedReportID}.lastMessageTimestamp`, 0);
+
+        // Determines if the active report has a history of draft comments while active.
+        // If the message has been sent, set the flag to false. Do not rely on nextProps.reportsWithDraft, which may still carry true in this update cycle.
+        const hasDraftHistory = isActiveReportSame && prevState.activeReport.lastMessageTimestamp !== lastMessageTimestamp ? false
+
+            // If the flag was true, preserve the state even if there is currently no draft. Otherwise, update it.
+            : (isActiveReportSame && prevState.activeReport.hasDraftHistory)
+                || lodashGet(nextProps.reportsWithDraft, `${ONYXKEYS.COLLECTION.REPORTS_WITH_DRAFT}${nextProps.currentlyViewedReportID}`, false);
```

Comment:
This just feels quite hard to read and understand what exactly is going in here, maybe it might be useful to split it up into longer version with more comments, which wont look so tidy but for other developers reading this code it might be quicker to understand in a more expanded version.

Thoughts?

Reply:
Hopefully, the comments are more on point now. Another way to put it:

```javascript
        // Determines if the active report has a history of draft comments while active.
        let hasDraftHistory;

        // If the active report has not changed and the message has been sent, set the draft history flag to false so LHN can reorder.
        // Otherwise, if the active report has not changed and the flag was previously true, preserve the state so LHN cannot reorder.
        // Otherwise, update the flag from the prop value.
        if (isActiveReportSame && prevState.activeReport.lastMessageTimestamp !== lastMessageTimestamp) {
            hasDraftHistory = false;
        } else if (isActiveReportSame && prevState.activeReport.hasDraftHistory) {
            hasDraftHistory = true;
        } else {
            hasDraftHistory = lodashGet(nextProps.reportsWithDraft, `${ONYXKEYS.COLLECTION.REPORTS_WITH_DRAFT}${nextProps.currentlyViewedReportID}`, false);
        }
```

Reply:
@marktoman Oh yeah, this reads better for sure! Can you please push these changes? Thank you very much!